### PR TITLE
#140 - add update and delete opinion features

### LINKED ISSUE #140: Add update and delete methods for opinions
For now opinions have only store method.

### PR DESCRIPTION
**Pull request by vojcc**
This should close #140.

### CODE REVIEW

File: lang/pl.json [hunk 1]
```diff
@@ -129,7 +129,10 @@
 
     "Opinion added successfully!": "Opinia dodana pomyślnie!",
     "There was an error adding your opinion!": "Wystąpił błąd podczas dodawania opinii!",
+    "Opinion edited successfully!": "Opinia zaktualizowana pomyślnie!",
```
Comment (EwelinaSkrzypacz) [inline lang/pl.json:132]: Please remove exclamation marks - we don't want to scream at user.

**Review by krzysztofrewak — APPROVED** [review]

**Review by AleksandraKozubal — APPROVED** [review]

**Review by EwelinaSkrzypacz — COMMENTED** [review]
1. Confirmation modal is not translated at all (only przycisk "Usuń")

image

2. "Pole content nie może być dłuższe niż 250 znaków." - word "content" is not translated

3. Maybe administrator should be able to delete all opinions? Or edit? We can make issue for that to discuss it and implement in the future.

**Comment by AleksandraKozubal** [comment]
@vojcc are you interested in finishing this pr?

**Comment by vojcc** [comment]
> @vojcc are you interested in finishing this pr?

@AleksandraKozubal, someone can take a wheel.

**Comment by AleksandraKozubal** [comment]
> > @vojcc are you interested in finishing this pr?
> 
> @AleksandraKozubal, someone can take a wheel.

@vojcc alright, @Lee0z is working on it.

**Comment by Lee0z** [comment]
@EwelinaSkrzypacz 

>Confirmation modal is not translated at all (only przycisk "Usuń")

image

I've managed to translate the header of modal and cancel button.

Content of this modal came out a little bit tricky. It has to be universal as component, but for now it doesn't work, WIP. 

Also found a bug when trying to delete city:
image

I translated the header, but that means every occurence of DeleteModal component needs to be revised. (Headers might be not translated). Should we make another issue? 

>"Pole content nie może być dłuższe niż 250 znaków." - word "content" is not translated

I can't find any proper solution to "Pole content nie może być dłuższe niż 250 znaków." for now.

>Maybe administrator should be able to delete all opinions? Or edit? We can make issue for that to discuss it and implement in the future

Sure, but it needs to be discussed in new issue.

**Review by EwelinaSkrzypacz — COMMENTED** [review]
@Lee0z 

1. I don't have translated text in delete modal (my application language is Polish):

image


We can leave this problem for another issue, but please crate one

2. You can translate word `content` in `lang/pl/validation.php` in array `atrributes`. 

3. Please make issue about managing opinions by administrator.

**Comment by EwelinaSkrzypacz** [comment]
> Didn't commit changes yesterday.
> 
> Translations are still work in progress, noticed new bug - after opinion deletion, toast is not translated. image

You can translate this in file `lang/pl.json`

**Comment by Lee0z** [comment]
Toast is now translated.

Fixed translation on DeleteModal content.
image

Any ideas how to universally translate content to match polish grammar?
image

**Review by EwelinaSkrzypacz — COMMENTED** [review]
Translation are almost fixed. Please look at screenshot below

image

It should be here "usunięte"

image

"Delete country" is not translated. It should be here "usunięty"

**Comment by Lee0z** [comment]
Maybe we should change DeleteModal's header to something like: 

"Usunąć miasto (city name)?
"Usunąć opinię?"
"Usunąć kraj (country name)? 

And in modal's content:
"Tej operacji nie można cofnąć"

This way, we'll avoid conflicting polish grammar.

**Comment by Lee0z** [comment]
Is it alright?:
- when deleting opinion:
image

- when deleting city:
image

- when deleting country:
image

added trimming translationKey in renderHeader to ensure there is no space before question mark.

**Review by EwelinaSkrzypacz — APPROVED** [review]
For me it's great :smile:

File: app/Importers/DataImporter.php [hunk 2]
```diff
@@ -59,18 +59,19 @@ public function translate(string $word, string $language): string
...
-            );
+        $cityAttributes = [
+            "city_name" => $cityName,
+            "country_name" => $countryName,
+        ];
+
+        $city = CityWithoutAssignedCountry::query()->withTrashed()->updateOrCreate(
+            $cityAttributes,
+            $cityAttributes,
+        );
```
Comment (kamilpiech97) [inline app/Importers/DataImporter.php:70]: What if I need to create a city but it is deleted (destroyed). Should this city be returned to the list of cities?

Comment (vojcc) [reply app/Importers/DataImporter.php:70]: But if you want to add city which is deleted you can. There's no correlation between `City` and `CityWithoutAssignedCountry`.

**Review by kamilpiech97 — APPROVED** [review]

**Comment by EwelinaSkrzypacz** [comment]
@Lee0z you can merge this pr :smiley:

**Pull request merged**